include/mbedtls/aes.h

```diff
@@ -57,68 +60,68 @@ extern "C" {
 #endif
 
 /**
- * \brief          AES context structure
- *
- * \note           buf is able to hold 32 extra bytes, which can be used:
- *                 - for alignment purposes if VIA padlock is used, and/or
- *                 - to simplify key expansion in the 256-bit case by
- *                 generating an extra round key
+ * The AES context-type definition. The AES context is passed to the APIs called.
+ * \note           This buffer can hold 32 extra Bytes, which can be used for one of the following purposes:
+ *                 <ul><li>Alignment if VIA padlock is used.</li>
+ *                 <li>Simplifying key expansion in the 256-bit case by generating an extra round key.</li></ul>
  */
 typedef struct
 {
-    int nr;                     /*!<  number of rounds  */
-    uint32_t *rk;               /*!<  AES round keys    */
-    uint32_t buf[68];           /*!<  unaligned data    */
+    int nr;                     /*!< The number of rounds.*/
+    uint32_t *rk;               /*!< AES round keys.*/
+    uint32_t buf[68];           /*!< Unaligned data buffer.*/
 }
 mbedtls_aes_context;
 
 /**
- * \brief          Initialize AES context
+ * \brief          This function initializes the specified AES context. To operate the AES machine, this must be the first API called.
  *
- * \param ctx      AES context to be initialized
+ * \param ctx      The AES context to initialize.
  */
 void mbedtls_aes_init( mbedtls_aes_context *ctx );
 
 /**
- * \brief          Clear AES context
+ * \brief          This function releases and clears the specified AES context.
  *
- * \param ctx      AES context to be cleared
+ * \param ctx      The AES context to clear.
  */
 void mbedtls_aes_free( mbedtls_aes_context *ctx );
 
 /**
- * \brief          AES key schedule (encryption)
+ * \brief          This function initializes the context set in the \p ctx parameter and sets the encryption key schedule for the AES operation.
```

Comment:
I made the modifications you wanted. However:
1. Brief means not a paragraph, but still a full sentence that describes what the function actually does.
2. In the ctx parameter, the description is "The AES context to initialize."  which is why I added that the function also initializes the context.

Reply:
A bit of context (no pun intended): the use of terminology around init(ialization)/free in the library documentation is not very clear. But it's consistent across the library. I do think that we need to improve this, but we should do this throughout the library in one shot, otherwise the internal inconsistency would be confusing.

To keep things consistent with the rest of the library, we should have used “set up” and not “initialize” here. We use “initialize” for the very first thing that happens to a context, preparing the empty box, and “set up” for putting data into the context. So the parameter description should be “the AES context to set up”. But the function description should focus on the purpose of the function, because “set up” is not idiomatic regarding AES. This function sets the key in the context. “AES key setup” could work as a brief description.

Reply:
Rephrased in light of your explanation